### ISSUE TITLE
Visualizer won't work if osu is in fullscreen mode.

### ISSUE THREAD

**Issue by spikeESP** [Piotrekol/StreamCompanion#50]
As the title says, the visualizer only updates when my osu! is either in borderless mode or in windowed mode. If I use fullscreen the visualizer just freezes in the last song it detected. np_all works fine, it's just the new visualizer. 

I've tried with both OBS Studio and Streamlabs OBS to see if any of them was the problem but it happens in both.

**Comment by Drewdememer** [comment]
It works correctly if game has compatibility mode enabled, but then game performance drops a lot (at least for me).

**Comment by spikeESP** [comment]
> It works correctly if game has compatibility mode enabled, but then game performance drops a lot (at least for me).

if it makes the game run worse then it's not a solution. But thanks, I'll try

**Comment by Piotrekol** [comment]
Currently I have no idea what is causing this (I can't reproduce this behavior either). 
From what I've been told on discord, placing visualizer window on second virtual screen makes it update as normal (win+tab on windows10)

**Comment by spikeESP** [comment]
> Currently I have no idea what is causing this (I can't reproduce this behavior either).
> From what I've been told on discord, placing visualizer window on second virtual screen makes it update as normal (win+tab on windows10)

I tried what Drewdememer said and used compatibility mode. Works fine now but some people will have their osu! running worse on said compatibility mode.

**Comment by Drewdememer** [comment]
> placing visualizer window on second virtual screen makes it update as normal

Sadly, didn't work for me.

**Comment by spikeESP** [comment]
> > placing visualizer window on second virtual screen makes it update as normal
> 
> Sadly, didn't work for me.

I was not very confident with it but for some reason compatibility mode is not adding me any input lag. As I said the only downside is that while streaming, If I alt- tab out, it looks weird. 

But I know from experience that to some, liek you, compatibility mode just makes the game unplayable.

**Comment by Drewdememer** [comment]
> compatibility mode just makes the game unplayable

Not really, i get ~1.2-1.4ms with it, but additional 0.4ms is pretty significant for me.

**Comment by spikeESP** [comment]
> > compatibility mode just makes the game unplayable
> 
> Not really, i get ~1-1.2ms with it, but additional 0.4ms is pretty significant for me.

I dont seem to have any at all. But I just discovered that sometimes getting back into the game takes a couple of tries.

**Comment by Drewdememer** [comment]
Okay, i discovered something. Visualizer doesn't update when it's somewhere on screen. If you drag it out of actual display like that https://drewdeautism.s-ul.eu/uhB8lPK8.png and then drag it back on the screen, it updates it https://drewdeautism.s-ul.eu/AueFIRGM.png

2nd virtual desktop doesn't work like that.

**Comment by spikeESP** [comment]
> Okay, i discovered something. Visualizer doesn't update when it's somewhere on screen. If you drag it out of actual display like that https://drewdeautism.s-ul.eu/uhB8lPK8.png and then drag it back on the screen, it updates it https://drewdeautism.s-ul.eu/AueFIRGM.png
> 
> 2nd virtual desktop doesn't work like that.

I don't get your point. I don't get the images.

**Comment by Piotrekol** [comment]
From the images I can deduct that underlying data is updating correctly but something blocks WPF window updates/redraws.

**Comment by Piotrekol** [comment]
@spikeESP @Drewdememer 
Can both of you provide your pc specs? (mainly interested in gpu - both integrated and dedicated regardless of with one is used)

**Comment by spikeESP** [comment]
> @spikeESP @Drewdememer
> Can both of you provide your pc specs? (mainly interested in gpu - both integrated and dedicated regardless of with one is used)

I'll give you full specs just in case:
Ryzen 5 1600X on a B450M
8GB RAM @2133Mhz
RX 570 8GB

**Comment by Drewdememer** [comment]
MSI H110M Pro-D
Intel Pentium G4400
nVidia GT 1030
4 GB RAM @2133Mhz

**Comment by Drewdememer** [comment]
But after update it works fine actually.

**Comment by Piotrekol** [comment]
Reproduced by having connected only one screen. 
Instead of window capture use game capture.

**Comment by spikeESP** [comment]
> [Reproduced by having only one screen connected](https://clips.twitch.tv/AmericanTenaciousOctopusPrimeMe)
> Instead of window capture use game capture.

will try.

**Comment by Unmeiu** [comment]
It's not working for me whatsoever. I've tried compatibility mode (latency is so bad it makes the game unplayable), I've tried disabling msn then re-installing stream companion, I've tried using game capture to capture the application as well. Sometimes it will switch the background for the map, and sometimes it will just do absolutely nothing. And even when the background and title decide to change, the pp counter doesn't work at all. However, these problems have only happened while osu! was in fullscreen mode, but when my game is in borderless or windowed mode it has unplayable latency issues. 

Someone plz help
